contracts/nft/erc721m/ERC721CMInitializableV1_0_1.sol

```diff
@@ -250,14 +260,22 @@ contract ERC721CMInitializableV1_0_0 is
         address royaltyReceiver,
         uint96 royaltyFeeNumerator
     ) external onlyOwner {
+        if (_setupLocked) {
+            revert ContractAlreadySetup();
+        }
+
         if (globalWalletLimit > maxMintableSupply) {
             revert GlobalWalletLimitOverflow();
         }
+
+        _setupLocked = true;
         _mintable = true;
         _maxMintableSupply = maxMintableSupply;
         _globalWalletLimit = globalWalletLimit;
         _mintCurrency = mintCurrency;
         _fundReceiver = fundReceiver;
+        _currentBaseURI = baseURI;
+        _tokenURISuffix = tokenURISuffix;
```

Comment:
Why do we need to store the suffix on-chain?

Reply:
I think this is because `tokenUri` is dynamically computed using `base`, `suffix` and `tokenId`.